### PR TITLE
Add details about completed operations in ClusterTopology

### LINKED ISSUE #14835: Add details about completed change in ClusterTopology
Part of #14462 
In the [spec for scaling api ](https://github.com/camunda/zeebe/pull/14816) we defined the response for querying ClusterTopology as follows:
```
{
  "version": 1,
  "brokers": [
    {
      "id": 8,
      ...
    }
  ],
  "change": {
    "id": 3,
    "status": "IN_PROGRESS",
    "startedAt": "2020-01-01T00:00:00Z",
    "completedAt": "2020-01-01T00:00:00Z", // only when pending is empty
    "internalVersion": 1,
    "completed": [
      {
        "operation": "BROKER_ADD",
        "brokerId": 8,
        "completedAt": "2020-01-01T00:00:00Z"
      }
    ],
    "pending": [
      ...
    ]
  }
}
```

To support this feature we have to keep track of the completed changes in `ClusterTopology`.  A list of completed and pending operations must be stored in ClusterChangePlan while the change is in progress. Once all operations are applied, both list can be cleared and it should store the info about last completed change containing id, status, startedAt and completedAt.

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

Restructured `ClusterChangePlan`  to add an optional completed change and optional pending change. PendingChange is updated while the change is in progress. Once the change is completed pending change is cleared and completed change will be updated with the id, started and completed timestamps.

## Related issues

closes #14835 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: topology/src/test/java/io/camunda/zeebe/topology/ClusterTopologyAssert.java [hunk 1]
```diff
@@ -86,4 +93,65 @@ public ClusterTopologyAssert hasVersion(final long version) {
...
+
+  /**
+   * Asserts that the actual topology has the same topology as the expected topology ignoring
+   * timestamps and versions.
+   */
+  public ClusterTopologyAssert hasSameTopologyAs(final ClusterTopology expected) {
+    assertThat(actual.members().keySet())
+        .containsExactlyInAnyOrderElementsOf(expected.members().keySet());
+
+    // Compare MemberStates without timestamp and version
```
Comment (oleschoenburg) [inline topology/src/test/java/io/camunda/zeebe/topology/ClusterTopologyAssert.java:105]: :wrench: I think this is easier to implement with something like
`assertThat(actual).usingRecursiveComparison().ignoringFields("timestamp", "version").isEqualTo(expected)`

File: topology/src/main/resources/proto/topology.proto [hunk 2]
```diff
@@ -68,4 +91,11 @@ enum State {
   LEFT = 4;
 }
 
+enum ChangeStatus {
+  CHANGE_STATUS_UNKNOWN = 0;
```
Comment (oleschoenburg) [inline topology/src/main/resources/proto/topology.proto:95]: :upside_down_face: Just `UNKOWN` would be cleaner, same as for `State` above.

**Review by lenaschoenburg — APPROVED** [review]
Changes look good but please consider my suggestion for slightly restructuring the types.

File: topology/src/main/resources/proto/topology.proto (hunk 2 above)
Comment (deepthidevaki) [reply topology/src/main/resources/proto/topology.proto:95]: If I remember correctly, protobuf was complaining when there two different enums in the same file with same name. Let me check again.

Comment (oleschoenburg) [reply topology/src/main/resources/proto/topology.proto:95]: Oh okay, how annoying :(

Comment (deepthidevaki) [reply topology/src/main/resources/proto/topology.proto:90]: `Note that enum values use C++ scoping rules, meaning that enum values are siblings of their type, not children of it.  Therefore, "UNKNOWN" must be unique within "topology_protocol", not just within "ChangeStatus".`

**Comment by deepthidevaki** [comment]
I have restructured `ClusterTopology` to include completed change in the topology instead of ClusterChangePlan. I will merge once the CI is green.

**Comment by deepthidevaki** [comment]
bors merge

**Pull request merged**